src/pixelator/utils.py

```diff
@@ -431,19 +431,26 @@ def is_read_file(read: str, read_type: Literal["r1"] | Literal["r2"]) -> bool:
     :raise ValueError: if the read file does not have a valid extension
     :raise AssertionError: if the read_type is not 'r1' or 'r2'
     """
+    read = Path(read).name
+
+    if read_type not in ("r1", "r2"):
+        raise AssertionError("Invalid read type: expected 'r1' or 'r2'")
+
     if not (read.endswith("fq.gz") or read.endswith("fastq.gz")):
         raise ValueError("Invalid file extension: expected .fq.gz or .fastq.gz")
 
     matches = []
     read_stem = Path(read.removesuffix(get_extension(read, 2)).rstrip(".")).name
     if read_type == "r1":
-        matches = re.findall(R"(.[Rr]1|(_[Rr]?1))", read_stem)
+        matches = re.findall(R"(.[Rr]1$)|(_[Rr]?1$)|(_[Rr]?1_[0-9]{3}$)", read_stem)
```

Comment:
Regarding this, it is for the Illumina numbered errors, right?

Reply:
Yes this is to support things like: `RDRS182520_S1_L001_R1_001.fastq.gz`. Now that we always match from the end of the string we need to add this 3 digit segment.